src/piece_picker.cpp

```diff
@@ -3076,7 +3076,7 @@ namespace libtorrent
 		TORRENT_ASSERT(m_seeds >= 0);
 		INVARIANT_CHECK;
 
-		avail.resize(m_piece_map.size());
```

Comment:
this is what I meant by preserving the original ``resize()`` function. I think this should work without a cast